### ISSUE TITLE
GPU Utilization很低

### ISSUE THREAD

**Issue by TJCVRS** [chineseocr/chinese-ocr#113]
想请问下作者使用的中文字典的数量大概是多少呢？　我使用了将近４０００个常用汉字生成训练数据，训练过程中不知道是否是由于全连接的输出(中文字典的长度)维度太高，导致迭代速度很慢，而且GPU Utilization很低，有时甚至接近于０

**Comment by wenlihaoyu** [comment]
5000多个字符。用keras训练还是pytorch?

**Comment by XufengLi** [comment]
我觉得是io读取的时间太长

**Comment by Sanster** [comment]
keras 版本的话用的是 tensorflow 的 ctc_loss 吧，这个好像没有 GPU 的实现，可能导致 GPU 利用率低。pytorch 版本用的是百度的 wrapctc，是有 GPU 实现的

**Comment by TJCVRS** [comment]
@Sanster tf的ctc loss并不慢 但是beam search会很慢 不知道是不是我用的有问题

**Comment by Sanster** [comment]
训练的时候可以不用跑 beam search 吧，或者不要每一步都跑

**Comment by TJCVRS** [comment]
@Sanster 现在已经关掉了 不过不知道为什么beam search会慢

**Comment by Sanster** [comment]
@TJCVRS beam search 确实挺慢的...今天刚发现，如果只是要解码，精度要求不高的话，可以就用 lstm 的输出接 softmax 来解码，速度快很多..

**Comment by TJCVRS** [comment]
@Sanster 我去试试 感谢
